site/app/libraries/database/DatabaseQueries.php

```diff
@@ -183,6 +183,24 @@ public function buildLoadThreadQuery($categories_ids, $thread_status, $show_dele
 
             $query_raw_join[] = "JOIN ({$query_select_categories}) AS QSC ON QSC.thread_id = t.id";
         }
+        // Unread Threads
+        if($unread_threads) {
+            $query_raw_where[] = 
+
+            "EXISTS(
+                SELECT thread_id 
+                FROM (posts LEFT JOIN forum_posts_history ON posts.id = forum_posts_history.post_id) AS jp
+                WHERE(
+                    jp.thread_id = t.id
+                    AND NOT EXISTS(
+                        SELECT thread_id 
+                        FROM viewed_responses v 
+                        WHERE v.thread_id = jp.thread_id
+                            AND v.user_id = ?
+                            AND (v.timestamp >= jp.timestamp
+                            AND (jp.edit_timestamp IS NULL OR (jp.edit_timestamp IS NOT NULL AND v.timestamp >= jp.edit_timestamp))))))";
+            $query_parameters[] = $current_user;
```

Comment:
As per comment below we might be able to simplify this. I will double check to see if edits should also be included.